python/lsst/obs/base/exposureAssembler.py

```diff
@@ -237,12 +229,19 @@ def assemble(self, components):
         info.setDetector(components.pop("detector", None))
         info.setTransmissionCurve(components.pop("transmissionCurve", None))
 
-        # Filter needs to be updated specially to match Exposure behavior
+        # A FilterLabel component gets read correctly even for version 1
+        # Exposure files, which persist the info as a Filter.
+        # TODO: deprecate in DM-27177, remove in DM-27811
+        info.setFilterLabel(components.pop("filterLabel", None))
```

Comment:
Shouldn't we be deriving `Filter` then from this `FilterLabel`? Is it okay to call `setFilter` and `setFilterLabel`?

Reply:
In principle you only need to call one (they're backed by the same store inside `Exposure`), but derivation is a bit tricky if `components` includes "filter" but not "filterLabel" (which, per the spec, is entirely legal). I think this depends on the bigger question above of how the "filter" component is implemented.

Reply:
Yes. Exactly that. derivedComponents. We shouldn't be storing the same information in two different components but if we can derive the second component from the first then that is the right thing to do.